mechanicalsoup/browser.py

```diff
@@ -37,6 +38,8 @@ def __init__(self, session=None, soup_config={'features': 'lxml'},
         self.__raise_on_404 = raise_on_404
         self.session = session or requests.Session()
 
+        self._finalize =  weakref.finalize(self.session,self.close)
```

Comment:
Nit: missing space after `,`. I guess flake8 will tell you when the Travis-CI build completes ;-).

Reply:
Oh, and extra-space after `=`.